src/Root.cc

```diff
@@ -389,6 +389,10 @@ Errors Root::Load(SDFPtr _sdf, const ParserConfig &_config)
   // Check that //axis*/xyz/@expressed_in values specify valid frames.
   checkJointAxisExpressedInValues(this, errors);
 
+  // Calculate Inertials for all worlds and models belonging to this root object
+  Errors inertialErrors = CalculateInertials(_config);
```

Comment:
Pose graphs have to be built before we can calculate composite inertials, so we can't call `Link::CalculateInertials`  in `Link::Load`.  We discussed putting in `Root::Load`, but we ran into issues where SDF objects being serialized and deserialized.

Having the inertia calculation be controlled by the user would add a bit of complexity, but it would have these advantages:
  * allow users to control whether computed inertials should be saved into the SDF for future xml output
  * update collision shapes and call `CalculateInertials` to recompute inertials

Reply:
> Pose graphs have to be built before we can calculate composite inertials, so we can't call `Link::CalculateInertials` in `Link::Load`.

ok, that's a good point

> We discussed putting in `Root::Load`, but we ran into issues where SDF objects being serialized and deserialized in gz-sim.

from a `libsdformat` perspective, I think it makes sense to put it in `Root::Load`, but it sounds like gz-sim doesn't handle that well. Having an additional method to call after `Load` reminds me of the `Init` methods in gazebo-classic. Could we add a flag to `Root::Load` / `ParserConfig` about whether to calculate inertials? then gz-sim could pass the flag telling it not to do so

Reply:
@scpeters That seems like a good idea. We were planning to add an Enum that can be passed to the `CalculateInertial()` function to configure its behavior (Save Calculations, Reset Calculations, etc.). 
But instead of that, we can add an Enum member to `ParserConfig` with `DONT_CALCULATE` as one of the values that would tell `Root::Load()` to not call the `CalculateInertial()` function. Does that work?

Reply:
yes that works!

Reply:
I had meant to say here that the SAVE behavior should be the default and that gz-sim should explicitly opt-in to SKIP, but I didn't fully articulate that. I've proposed this behavior change in https://github.com/gazebosim/sdformat/pull/1325 and https://github.com/gazebosim/gz-sim/pull/2166